### PR TITLE
Fix OLM E2E Tests

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description 

Visit the detail view for the `CatalogSource` and create subscription from there instead of the list view.

### CODE REVIEW

File: frontend/integration-tests/tests/olm/descriptors.scenario.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ const defaultValueFor = <C extends SpecCapability | StatusCapability>(capability
   }
 };
 
-xdescribe('Using OLM descriptor components', () => {
+describe('Using OLM descriptor components', () => {
```
Comment (alecmerdler) [inline frontend/integration-tests/tests/olm/descriptors.scenario.ts]: Whoops

File: frontend/integration-tests/views/catalog.view.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ import { browser, $, $$, by, ExpectedConditions as until, ElementFinder } from '
 
 const moveTo = (elem: ElementFinder) => browser.actions().mouseMove(elem).perform().then(() => elem);
 
+const withRetry = (action) => (args) => action(args)
+  .catch(() => action(args))
+  .catch(() => action(args))
+  .catch(() => action(args));
```
Comment (spadgett) [inline frontend/integration-tests/views/catalog.view.ts]: Do we know why this is needed? I'd prefer to fix the underlying cause.

Comment (spadgett) [reply frontend/integration-tests/views/catalog.view.ts]: We get the same error sometimes in the annotation tests, too. I suspect we're iterating over DOM elements that are replaced in the page. Maybe it's the use of `filter` in `entryRowFor`?

File: frontend/integration-tests/views/catalog.view.ts [hunk 3]
```diff
@@ -17,6 +22,10 @@ export const hasSubscription = (name: string) => browser.getCurrentUrl().then(ur
   return entryRowFor(name).element(by.buttonText('Create Subscription')).isPresent();
 }).then(canSubscribe => !canSubscribe);
 
-export const createSubscriptionFor = (pkgName: string) => moveTo(
-  entryRowFor(pkgName).element(by.buttonText('Create Subscription'))).then(el => el.click()
-);
+export const viewCatalogDetail = (name: string) => $$('.co-catalogsource-list__section').filter(section => section.$('h3').getText()
```
Comment (spadgett) [inline frontend/integration-tests/views/catalog.view.ts]: Does something like this work instead? (untested)

```js
element(by.cssContainingText('.co-catalogsource-list__section h3', name)).element(by.linkText('View catalog details')).click();
```

Comment (spadgett) [reply frontend/integration-tests/views/catalog.view.ts]: Ugh, never mind. That won't work.

**Comment by alecmerdler** [comment]
jenkins rebuild

**Comment by alecmerdler** [comment]
@spadgett It's passed CI 3 times now with these changes, we might be good.

**Comment by alecmerdler** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
Merging to unblock CI, although we might want to revisit this.

/lgtm

**Comment by spadgett** [comment]
Heh, same problem in the annotation tests. We'll want to look at what causes this and see if we can't fix later on.

```
1) Modal Annotations : Add numeric Annotation from grid
   StaleElementReferenceError: stale element reference: element is not attached to the page document
```

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
Opened https://jira.coreos.com/browse/CONSOLE-876

**Comment by spadgett** [comment]
```
1) Using OLM descriptor components : displays list containing custom resource
   Error: Command failed: echo '{"apiVersion":"apiextensions.k8s.io/v1beta1","kind":"CustomResourceDefinition","metadata":{"name":"apps.test.tectonic.com","labels":{"automatedTestName":"test-qmwvd"}},"spec":{"group":"test.tectonic.com","version":"v1","scope":"Namespaced","names":{"plural":"apps","singular":"app","kind":"App","listKind":"Apps"}}}' | kubectl create -f -
   Error from server (AlreadyExists): error when creating "STDIN": customresourcedefinitions.apiextensions.k8s.io "apps.test.tectonic.com" already exists
```

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**